src/main/java/picocli/CommandLine.java

```diff
@@ -5017,6 +5048,67 @@ public interface ITypeConverter<K> {
     }
 
+    /**
+     * A pluggable converter for enums.
+     * <p>
+     * This interface defines the contract for classes that know how to convert a String
+     * into enum types in general. Custom converters can be registered with the
+     * {@link CommandLine#registerEnumConverter(IEnumConverter)} method.
+     * </p>
+     * @see ITypeConverter
+     */
+    public interface IEnumConverter {
+        /**
+         * Converts the specified command line argument value to some enum.
+         * @param value the command line argument String value
+         * @param enumType the class representing the type of enum
+         * @param caseInsensitiveEnumValuesAllowed whether the parser should ignore case when converting arguments to {@code enum} values
+         * @return the resulting enum
+         * @throws Exception an exception detailing what went wrong during the conversion.
+         *          Any exception thrown from this method will be caught and shown to the end user.
+         *          An example error message shown to the end user could look something like this:
+         *          {@code Invalid value for option '--some-option': cannot convert 'xxxinvalidinput' to SomeEnumType (java.lang.IllegalArgumentException: Invalid format: must be 'x:y:z' but was 'xxxinvalidinput')}
+         * @throws TypeConversionException throw this exception to have more control over the error
+         *          message that is shown to the end user when type conversion fails.
+         *          An example message shown to the user could look like this:
+         *          {@code Invalid value for option '--some-option': Invalid format: must be 'x:y:z' but was 'xxxinvalidinput'}
+         */
+        <E extends Enum<E>> E convert(String value, Class<E> enumType, boolean caseInsensitiveEnumValuesAllowed) throws Exception;
+    }
+
+    /**
+     * The enum converter installed by default. Custom enum converters can replace
+     * the default enum converter altogether, or selectively convert certain enums
+     * and then fall back to the default behavior by delegating to this default implementation.
+     */
+    public static final IEnumConverter DEFAULT_ENUM_CONVERTER = new IEnumConverter() {
+        @Override
+        public <E extends Enum<E>> E convert(String value, Class<E> enumType, boolean caseInsensitiveEnumValuesAllowed) throws Exception {
+            try { return Enum.valueOf(enumType, value); }
+            catch (IllegalArgumentException ex) {
+                for (E enumConstant : enumType.getEnumConstants()) {
+                    String str = enumConstant.toString();
+                    String name = enumConstant.name();
+                    if (value.equals(str) || value.equals(name) || caseInsensitiveEnumValuesAllowed && (value.equalsIgnoreCase(str) || value.equalsIgnoreCase(name))) {
+                        return enumConstant;
+                    }                            }
```

Comment:
I just noticed this formatting oddity and thought I had made a typo, but then I saw it was in the original code, so I'll leave it as-is for the moment.

Reply:
I noticed it too in the original code. Looks like a newline was accidentally deleted. It was not intentional, please fix in the PR.